fcli-core/fcli-ssc/src/main/java/com/fortify/cli/ssc/token/helper/SSCTokenConverter.java

```diff
@@ -36,11 +39,18 @@ public static final char[] toRestToken(char[] token) {
     }
     
     public static final boolean isApplicationToken(String token) {
-        return token.matches("^[\\da-f]{8}(?:-[\\da-f]{4}){3}-[\\da-f]{12}$");
+        return applicationTokenPattern.matcher(token).matches();
     }
     
     private static final String decode(String token) {
-        return new String(Base64.decodeBase64(token), StandardCharsets.UTF_8);
+        return validateApplicationTokenFormat(new String(Base64.decodeBase64(token), StandardCharsets.UTF_8));
```

Comment:
Looks like this method is now only called when decoding a REST token. If the given token is already a REST token, the `toRestToken` method returns the given token as-is, without performing any validation (decode & validate decoded token).